weblate/utils/tasks.py

```diff
@@ -83,25 +83,45 @@ def database_backup():
         return
     ensure_backup_dir()
     database = settings.DATABASES["default"]
-    if database["ENGINE"] != "django.db.backends.postgresql":
-        return
-    cmd = ["pg_dump", "--dbname", database["NAME"]]
-    if database["HOST"]:
-        cmd += ["--host", database["HOST"]]
-    if database["PORT"]:
-        cmd += ["--port", database["PORT"]]
-    if database["USER"]:
-        cmd += ["--username", database["USER"]]
-    if settings.DATABASE_BACKUP == "compressed":
-        cmd += ["--file", data_dir("backups", "database.sql.gz")]
-        cmd += ["--compress", "6"]
+    if database["ENGINE"] == "django.db.backends.postgresql":
```

Comment:
```suggestion
    env = get_clean_env()
    if database["ENGINE"] == "django.db.backends.postgresql":
```

Call the function just once and update the env later.